docs/json_schemas/kratos_external_idp/v0/provider.json

```diff
@@ -0,0 +1,132 @@
+{
+  "title": "ProviderSchema",
+  "description": "Provider schema for KratosExternalIdp.",
+  "type": "object",
+  "properties": {
+    "unit": {
+      "$ref": "#/definitions/BaseModel"
+    },
+    "app": {
+      "$ref": "#/definitions/KratosExternalIdpProviderData"
+    }
+  },
+  "required": [
+    "app"
+  ],
+  "definitions": {
+    "BaseModel": {
+      "title": "BaseModel",
+      "type": "object",
+      "properties": {}
+    },
+    "SecretBackend": {
+      "title": "SecretBackend",
+      "description": "An enumeration.",
+      "enum": [
+        "relation",
+        "secret",
+        "vault"
+      ]
+    },
+    "Provider": {
+      "title": "Provider",
+      "description": "An enumeration.",
+      "enum": [
+        "microsoft",
+        "apple",
+        "generic",
+        "auth0",
+        "google",
+        "facebook",
+        "github",
+        "gitlab",
+        "slack",
+        "spotify",
+        "discord",
+        "twitch",
+        "netid",
+        "yandex",
+        "vk",
+        "dingtalk"
```

Comment:
Do we actively support all of these right now (as in: do we test them?).

If not, I'd suggest adding just those we're confident work

Reply:
The only ones that we test are `microsoft` and `generic`. I agree to limit them to the ones that we are certain can work, but we probably would like to allow for a list bigger than those 2. 

@massigori which ones should I keep?